test/gurka/test_turn_lanes.cc

```diff
@@ -393,6 +393,145 @@ TEST(Standalone, TurnLanesUTurns) {
+       {"LP", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "First Street"}}},
+       {"MN", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Second Street"}}},
+       {"GB", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+       {"BJ", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+       {"KC", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+       {"CF", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}}};
+
+  const gurka::relations relations = {
+      {{
+           {gurka::way_member, "CF", "from"},
+           {gurka::way_member, "GB", "to"},
+           {gurka::way_member, "FG", "via"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_u_turn"},
+       }},
+      {{
+           {gurka::way_member, "FL", "from"},
+           {gurka::way_member, "IG", "to"},
+           {gurka::way_member, "LI", "via"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_u_turn"},
+       }},
+  };
+
+  const auto layout =
+      gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+
+  auto map = gurka::buildtiles(layout, ways, {}, relations, "test/data/gurka_turn_lanes_7");
+
+  valhalla::Api result;
+
+  // Test U-turn using left-only lanes. The left-most left lane should be
+  // active & rest invalid
+  result = gurka::route(map, "A", "H", "auto");
+
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kUturnLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result, {
+                                  {5, "[ *left* ACTIVE | left | through | right | right ]"},
+                                  {5, "[ left | left | *through* ACTIVE | right | right ]"},
+                                  {0, ""},
+                                  {0, ""},
+                                  {0, ""},
+                              });
+
+  // Test using reverse lanes. The reverse lane should be active & the rest
+  // invalid
+  result = gurka::route(map, "E", "D", "auto");
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kUturnLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result, {
+                                  {4, "[ *reverse* ACTIVE | left | through | right ]"},
+                                  {4, "[ *reverse* ACTIVE | left | through | right ]"},
+                                  {0, ""},
+                                  {5, "[ left | left | *through* ACTIVE | right | right ]"},
```

Comment:
I'm not sure but I think this might be on edge BC which has `{"turn:lanes", "left|left|through|right|right"}` from way ABC. If that's the case, I guess it's correct?

Reply:
isn't this also an artifact of the bug you explained above? i.e the turn lane info on internal intersection `C` should be overridden by that from `EF`. The internal intersection is only an artifact of digitization, so a driver would never see the lanes from `ABC`, right? This route is making a left-uturn, so I would expect only lane info from `EF` to be considered.